runtime/caml/sync.h

```diff
@@ -0,0 +1,92 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*          Xavier Leroy and Damien Doligez, INRIA Rocquencourt           */
```

Comment:
This is a new file implemented by @ctk21. The copyright header should be fixed appropriately.

Reply:
I'm not so sure, I essentially moved the code that previously was in systhreads (e.g. `st_posix.h`). My changes were really to do with naming.